### PR TITLE
Properly clip the range in rangeFromLineNumber

### PR DESCRIPTION
**Pull request by steelbrain**
Before

After

This is most likely gonna be an API breaking change, I am saying that because this broke linter-eslint's custom range patching :facepalm:

The upgrade commit for it is https://github.com/steelbrain/linter-eslint/commit/2b8ba8f5060c75e362f09258c3a61808ba20674f

Todo
 - [x] specs
 - [ ] ~~docs~~

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -43,9 +43,16 @@ export function rangeFromLineNumber(textEditor: TextEditor, line: number, column
     throw new Error(`Column start (${colStart}) greater than line length (${lineLength})`)
   }
 
+  let colEnd = lineLength
+  const rowText = buffer.lineForRow(lineNumber).substr(colStart)
+  const match = /^( +|[\$\w]+)/.exec(rowText)
```
Comment (Arcanemagus) [inline src/index.js]: Just so our earlier conversation gets recorded, this line should use something along the lines of:

```js
textEditor.getCursors()[0].wordRegExp()
```

**Comment by steelbrain** [comment]
@Arcanemagus Ready for review round 2

File: spec/helper-spec.js [hunk 2]
```diff
@@ -48,7 +48,7 @@ describe('linter helpers', function () {
         expect(range[0][0]).toEqual(7)
```
Comment (Arcanemagus) [inline spec/helper-spec.js]: Why not condense these down to a `toEqual(` like the others? (Not necessary, but might as well be consistent)

File: src/index.js [hunk 3]
```diff
@@ -27,25 +27,29 @@ export function rangeFromLineNumber(textEditor: TextEditor, line: number, column
...
 
-  let colStart = column
-  if (!Number.isFinite(colStart) || Number.isNaN(colStart) || colStart < 0) {
-    const indentation = buffer.lineForRow(lineNumber).match(/^\s+/)
-    if (indentation && indentation.length) {
-      colStart = indentation[0].length
-    } else {
-      colStart = 0
-    }
+  const columnGiven = typeof column === 'number' && Number.isFinite(column) && column > -1
```
Comment (Arcanemagus) [inline src/index.js]: Why did you change the "check if it's a number" logic? If this is somehow better, then the logic on line 19 should be updated as well.

File: src/index.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ import { exec, execNode } from 'sb-exec'
 let NamedRegexp = null
 export const FindCache = new Map()
 
-export function rangeFromLineNumber(textEditor: TextEditor, line: number, column: number): Range {
+export function rangeFromLineNumber(textEditor: TextEditor, line: number, column: ?number): Range {
```
Comment (Arcanemagus) [inline src/index.js]: Since this function can accept just a `textEditor` as an argument, should `line` be listed as optional as well?

File: src/index.js [hunk 5]
```diff
@@ -27,25 +27,29 @@ export function rangeFromLineNumber(textEditor: TextEditor, line: number, column
...
+    lineText = lineText.substr(column)
   }
 
-  const lineLength = buffer.lineLengthForRow(lineNumber)
-
-  if (colStart > lineLength) {
-    throw new Error(`Column start (${colStart}) greater than line length (${lineLength})`)
+  let colEnd = lineLength
+  let colStart = columnGiven ? column : 0
+  const rowText = buffer.lineForRow(lineNumber).substr(colStart)
```
Comment (Arcanemagus) [inline src/index.js]: You're duplicating the logic from lines 31-36... aren't the lines below meant to just use `lineText`?

**Comment by Arcanemagus** [comment]
Round 3 😉.

File: src/index.js (hunk 5 above)
Comment (steelbrain) [inline src/index.js]: They are, I added the lines above later and forgot to update the ones below, good 👀

**Comment by steelbrain** [comment]
Taken care of 😉

**Comment by steelbrain** [comment]
CI is failing due to `apt-get update` issue unrelated to this PR, here's the local results

**Comment by Arcanemagus** [comment]
LGTM
